### PR TITLE
add help text for the `where` keyword

### PR DESCRIPTION
**Pull request by JeffBezanson**
Part of #22153, but not sure if more work is needed related to search.

### CODE REVIEW

**Review by ararslan — APPROVED** [review]
💯

**Review by StefanKarpinski — APPROVED** [review]

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -645,6 +645,36 @@ to be set after construction. See `struct` and the manual for more information.
...
+
+There is also a concise syntax for nested `where` expressions. For example, this:
+
+    Pair{T, S} where S<:Array{T} where T<:Number
+
+can be shortened to:
+
+    Pair{T, S} where {T<:Number, S<:Array{T}}
+
+This form is often found on method signatures.
```
Comment (Sacha0) [inline base/docs/basedocs.jl]: "found on" or "found in"?

**Review by Sacha0 — APPROVED** [review]

File: base/docs/basedocs.jl (hunk 1 above)
Comment (JeffBezanson) [inline base/docs/basedocs.jl]: I was thinking "on" as in "attached to", but yes I think any of those would be ok.

Comment (Sacha0) [inline base/docs/basedocs.jl]: Cheers, "on", "attached to", or what have you all sound good then :).

**Pull request merged**